code/game/objects/items/tools/maintenance_tools.dm

```diff
@@ -199,7 +199,22 @@
 			var/mob/living/L = O
 			L.IgniteMob()
 
-
+/obj/item/tool/weldingtool/use_tool(atom/target, mob/living/user, delay, amount, volume, datum/callback/extra_checks)
+	if(istype(target, /obj/machinery/door/airlock/multi_tile))
+		if((target.dir == 1) || (target.dir == 2))
+			target.add_overlay(GLOB.welding_sparks_multitiledoor_vertical)
+		else
+			target.add_overlay(GLOB.welding_sparks_multitiledoor_horizontal)
+	else
+		target.add_overlay(GLOB.welding_sparks)
+	. = ..()
+	if(istype(target, /obj/machinery/door/airlock/multi_tile))
+		if((target.dir == 1) || (target.dir == 2))
```

Comment:
A) dirs should use proper defines (ex: 1 = NORTH)
B) why not use & (ie thing.dir & NORTH)